src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1470,36 +1468,31 @@ export function registerTerminalActions() {
 			if (customType) {
 				return commandService.executeCommand(customType.command);
 			}
+			terminalService.configHelper.setWorkspaceShellAllowed(false);
 			if (detectedProfiles) {
-				let launchConfig = detectedProfiles?.find((profile: { profileName: string; }) => profile.profileName === profileSelection);
-				if (launchConfig && !launchConfig.isWorkspaceProfile) {
-					const instance = terminalService.createTerminal({ executable: launchConfig.path, args: launchConfig.args });
-					terminalService.setActiveInstance(instance);
-				} else if (launchConfig && launchConfig.isWorkspaceProfile) {
-					if (!storageService.getBoolean(`${QUICK_LAUNCH_PROFILE_CHOICE}-${item}-${launchConfig.path}`, StorageScope.WORKSPACE, false)) {
-						notificationService.prompt(Severity.Info, `Do you allow this workspace to modify your terminal profile? ${item} with path ${launchConfig?.path}`,
+				const launchConfig = detectedProfiles?.find((profile: { profileName: string; }) => profile.profileName === profileSelection);
+				if (launchConfig) {
+					if (!terminalService.configHelper.checkWorkspaceShellPermissions() || !launchConfig.isWorkspaceProfile) {
```

Comment:
`checkWorkspaceShellPermissions` won't work like this because it will only check if there are customizations to `shell` and `shellArgs`. But you're right that we should be using that function. I think we need to pass in to it whether we're checking a `profile` or `shell`/`shellArgs`.

This is what I'm thinking:

```ts
checkWorkspaceShellPermissions(profile?: ITerminalProfile)
```

Reply:
`isWorkspaceProfile` already checks this

Reply:
https://github.com/microsoft/vscode/blob/9093f265a64f640c3fb385195f6b1883deea2b93/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L298-L298